soaktest/soaktest.go

```diff
@@ -145,16 +146,16 @@ func rttStats(raw map[string]interface{}, topic string) error {
 		}
 		nodeid = int(nid)
 
-		tags := []string{fmt.Sprintf("broker:%d, type:%s, topic:%s, parts:%s",
-			nodeid, raw["type"], topic, parts)}
+		tags := []string{fmt.Sprintf("broker:%d, type:%s, parts:%s",
```

Comment:
If topparsValue.(map[string]interface{}) doesn't have any values, the parts will be empty, partitions won't be part of the per-broker rtts. But if there is value at topparsValue.(map[string]interface{}), this will contribute to the series for partition leaders.

Reply:
rtt is per broker though, not per partition, and if the partitions are part of the tags it means that any partition leader change will make a new time series for the broker, even though it is the same one.
timeseries 1: "broker:0, parts:1,2"
timeseries 2: "broker:0, parts:1"

But it is still the same broker and should be the same time series, so remove the partitions from the tags.

Only, I'm still not convinced that the tags should be concatenated in a single string like that, I think each key:value should be its own tag in that array.
E.g.: `[]string{"broker:0", "type:foo"}`